service/src/main/java/uk/gov/hmcts/reform/fpl/service/DirectionHelperService.java

```diff
@@ -0,0 +1,189 @@
+            .collect(groupingBy(directionElement -> directionElement.getValue().getAssignee().getValue()));
+    }
+
+    /**
+     * Iterates over a list of directions and adds numbers to the type starting from 2.
+     *
+     * @param directions a list of directions.
+     * @return a list of directions with numbered type.
+     */
+    public List<Element<Direction>> numberDirections(List<Element<Direction>> directions) {
+        AtomicInteger at = new AtomicInteger(2);
+
+        return directions.stream()
+            .map(direction -> {
+                Direction.DirectionBuilder directionBuilder = direction.getValue().toBuilder();
+
+                return Element.<Direction>builder()
+                    .id(direction.getId())
+                    .value(directionBuilder.type(at.getAndIncrement() + ". " + direction.getValue().getType()).build())
+                    .build();
+            })
+            .collect(toList());
+    }
+
+    /**
+     * Takes a direction from a configuration file and builds a CCD direction.
+     *
+     * @param direction  the direction taken from json config.
+     * @param completeBy the date to be completed by. Can be null.
+     * @return Direction to be stored in CCD.
+     */
+    public Element<Direction> constructDirectionForCCD(DirectionConfiguration direction, LocalDateTime completeBy) {
+        return Element.<Direction>builder()
+            .id(randomUUID())
+            .value(Direction.builder()
+                .type(direction.getTitle())
+                .text(direction.getText())
+                .assignee(direction.getAssignee())
+                .directionRemovable(booleanToYesOrNo(direction.getDisplay().isDirectionRemovable()))
+                .readOnly(booleanToYesOrNo(direction.getDisplay().isShowDateOnly()))
+                .completeBy(completeBy)
+                .build())
+            .build();
+    }
+
+    private String booleanToYesOrNo(boolean value) {
+        return value ? "Yes" : "No";
+    }
+
+
+    private List<Element<Direction>> assignCustomDirections(List<Element<Direction>> directions,
+                                                            DirectionAssignee assignee) {
+        return directions.stream()
+            .map(element -> Element.<Direction>builder()
+                .id(element.getId())
+                .value(element.getValue().toBuilder()
+                    .assignee(assignee)
+                    .custom("Yes")
+                    .readOnly("No")
+                    .build())
+                .build())
+            .collect(toList());
+    }
+
+    private List<Element<Direction>> filterDirectionsNotRequired(List<Element<Direction>> directions) {
+        return directions.stream()
+            .filter(directionElement -> directionElement.getValue().getDirectionNeeded() == null
+                || directionElement.getValue().getDirectionNeeded().equals("Yes"))
+            .collect(toList());
```

Comment:
could we have a situation where `directionElement.getValue().getDirectionNeeded()` is empty?

Reply:
directionNeeded is always populated as Yes / No, so there should never be instances of null or empty, have removed the first condition so that only = Yes remains 👍

Reply:
Direction is always null, Yes or No